src/dafny/BoundedInts.md

```diff
@@ -0,0 +1,51 @@
+
+## The `BoundedInts` module {#sec-boundedints}
+
+The `Dafny.BoundedInts` module contains definitions of types and constants for fixed-bit-width integers.
+
+Dafny itself generally uses and reasons about mathematical, unbounded integers and natural numbers.
+However compiled programs, for efficiency of computation and storage, will generally use fixed-bit-width
+numbers. So Dafny defines them here, for use when needed.
+
+An important implementation point about these type definitions is Dafny can determine which native type 
+in the target compiled program best matches the Dafny type. For example, the Dafny type `int16` is
+a signed integer that fits in 16 bits. If a program using this type is compiled to, say Java, then
+variables of this Dafny type will be compiled to Java `short` variables. In some cases there is no
+natural match. For example Java does not have an unsigned 16-bit type while C# and C++ do.
```

Comment:
YES, thank you! This is worth documenting somewhere with very high visibility in the reference manual/user guide as well.

Reply:
Good